contracts/interface/ScoringFunction.sol

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+interface IScoringFunction {
+    function calculateScore(uint256 distance, uint96 stake)
+        external returns (uint32);
+    function calculateXORDistance(uint8 hashingFunctionId, bytes memory nodeId, bytes memory keyword)
```

Comment:
I'd rename to calculateDistance as future distance calculations might not necessarily be an implementation of XOR (for every algorithm)

Reply:
Done